src/yang/ietf-bgp.yang

```diff
@@ -858,6 +858,54 @@ module ietf-bgp {
             "Clear neighbors action.";
 
           input {
+	    choice operation {
+	      default operation-admin;
+	      description
+	        "The type of operation for the clear action.";
+	      case operation-admin {
+	        leaf admin {
+		  type empty;
+		  description
+		    "Closes the Established BGP session with a BGP
+		     NOTIFICATION message with the Administrative
+		     Reset error subcode.";
+		  reference
+		    "RFC 4486.";
+		}
+	      }
+	      case operation-hard {
+	        leaf hard {
+		  type empty;
+		  description
+		    "Closes the Established BGP session with a BGP
+		     NOTIFICATION message with the Hard Reset error
+		     subcode.";
+		  reference
+		    "RFC 8538, Section 3.";
+		}
+	      }
+	      case operation-soft {
+	        leaf soft {
+		  type empty;
+		  description
+		    "Re-sends the current Adj-Rib-Out to this
+		     neighbor.";
+		}
+	      }
+	      case operation-soft-inbound {
+	        leaf soft-inbound {
+		  // XXX JMH - if we wanted to permit future expansion
+		  // to be per-AFI/SAFI, is this a correct definition?
```

Comment:
I am not clear on what the description of leaf 'soft-inbound' is? Is of type 'afi-safi-type' or 'empty'?

Also is the entire case of 'operation-soft-inbound' controlled by the feature statement, or just the setting of 'soft-inbound'?

Reply:
Ahh! The confusion was around the XXX comment.

And now that I understand the issue, the augmentation would have to be to the case statement to add another leaf that indicates the afi/safi type in addition to the existing leaf. 

HTH.

Reply:
Okay, additional leaves with the presence node in the same choice is a reasonable way this can be extended.  I'll remove the XXX and commit.